_api/hasura/cron/syncCoSouls.ts

```diff
@@ -216,24 +229,118 @@ const syncCoSoulToken = async (
   );
 };
 
-async function updateCoSoulOnChain(cosoul: CoSoul, totalPGIVE: number) {
+const syncPatchCoSoulToken = async (
+  updatedCosouls: { cosoul: CoSoul; localPGIVE: number }[]
+) => {
+  let payload = paddedHex(PGIVE_SLOT, 2, true); //1byte for slot
+  const successMessages = [];
+  for (const { cosoul, localPGIVE } of updatedCosouls) {
+    if (localPGIVE > 0) {
+      const pGIVE = Math.floor(localPGIVE);
+      //four bytes for pgive and four bytes for tokenId
+      payload += getPayload(pGIVE, cosoul.token_id);
+      successMessages.push(
+        `set PGIVE on chain for tokenId: ${cosoul.token_id} address: ${cosoul.address} to ${localPGIVE}`
+      );
+    } else {
+      console.log(
+        'skipping setting on-chain PGIVE because it is 0 for tokenId: ' +
+          cosoul.token_id +
+          ' address: ' +
+          cosoul.address
+      );
+    }
+  }
+  if (payload.length > 4) {
+    // 0x and one byte for slot 0x00
+    //if only one needs to be updated on chain, we can use setOnChainPGIVE
+    if (payload.length === 20) {
+      //0x00 plus 4 bytes for pgive and 4 bytes for tokenId
+      const tx = await setOnChainPGIVE(
+        updatedCosouls[0].cosoul.token_id,
+        updatedCosouls[0].localPGIVE
+      );
+      await tx.wait();
+    } else {
+      const tx = await setBatchOnChainPGIVE(payload);
+      await tx.wait();
+    }
+    successMessages.forEach(msg => console.log(msg));
+  }
+
+  await Promise.allSettled(
+    updatedCosouls.map(async ({ cosoul, localPGIVE }) => {
+      try {
+        await adminClient.mutate(
+          {
+            update_cosouls_by_pk: [
+              {
+                pk_columns: {
+                  id: cosoul.id,
+                },
+                _set: {
+                  pgive: localPGIVE,
+                  checked_at: new Date().toISOString(),
```

Comment:
for the future, these can just be `now()` like https://github.com/coordinape/coordinape/blob/main/api-lib/poap/poap-api.ts#L208